Gems/EMotionFX/Code/Source/Integration/Editor/Components/EditorActorComponent.cpp

```diff
@@ -1084,6 +1086,9 @@ namespace EMotionFX
                 return;
             }
 
+            // Remember the parent entity before we re-parent (attach) it.
+            AZ::TransformBus::EventResult(m_attachmentPreviousParent, GetEntityId(), &AZ::TransformBus::Events::GetParentId);
```

Comment:
I'm not very familiar with how the EditorActorComponent works but based on my understanding of O3DE's entity component system, you wouldn't need to reparent entities if all you are doing is modifying some component properties. I see that you are caching the previous parent to set it correctly later but my question is why is reparenting needed at all to begin with?

Reply:
That is a good point and I've played around with that: This is actually not needed for attachments using the attachment component, but for skinned attachments. Skinned attachments also need the entity transform to be driven so that the actor instance moves along with that. Skinned attachments don't use the attachment component but is some special setting in the actor component. I'm afraid this is a bigger thing to tackle and out of scope for this bug fix. When doing that, we should directly consolidate some code between the actor and the editor actor component, as we got some duplicated code here due to some tech debt/decision made in the past.

Reply:
Okay. Approving this since it looks like removing the transform dependency seems like a bit more involved here. I recommend revisiting the skinned attachments logic so that the reparenting logic isn't hardwired into the component logic itself but instead decoupled from it. For example, we can put some guardrails so that skinned attachments can only work if targeting existing parent. Not sure if that'll get the desired ux but just throwing some ideas out there...